src/foundation/y_check_base.clas.abap

```
@@ -636,13 +628,29 @@ CLASS y_check_base IMPLEMENTATION.
 
 
   METHOD raise_error.
+    IF clean_code_exemption_handler->is_object_exempted( object_name = object_name object_type = object_type ) = abap_true.
+      RETURN.
+    ENDIF.
+
+    TRY.
+        DATA(main_app_comp) = y_code_pal_app_comp=>get( ref_scan_manager->levels[ level = 0 ] ).
+        DATA(curr_app_comp) = y_code_pal_app_comp=>get( ref_scan_manager->levels[ statement_level ] ).
+
+        IF main_app_comp <> curr_app_comp.
+          RETURN.
+        ENDIF.
+      CATCH cx_sy_itab_line_not_found
+            ycx_entry_not_found.
```

empty catch. Move the logic to a method that returns a boolean.